File: mrc-unity/Assets/Scripts/Models/Member.cs
using System;
using System.Collections;
using System.Collections.Generic;
using UnityEngine;

public class Member
{
    public string username; // 사용자 ID
    public string nickname; // 사용자 닉네임
    public Cart selectedCart; // 선택된 카트
    public List<Cart> carts; // 사용자의 카트 목록
    public List<Record> records; // 사용자의 기록 목록
    public AccountType accountType;
    public string iconUrl;
    public int coin;

    // 초기화
    public void InitializeUser(string _username, string _nickname, List<Cart> _carts, Cart _selectedCart, List<Record> _records, AccountType _accountType, string _iconUrl, int _coin)
    {
        username = _username;
        nickname = _nickname;
        selectedCart = _selectedCart;
        carts = _carts;
        records = _records;
        accountType = _accountType;
        iconUrl = _iconUrl;
        coin = _coin;
    }

    // 카트 추가
    public void AddCart(Cart cart)
    {
        carts.Add(cart);
    }

    // 기록 업데이트
    public void UpdateRecord(List<Record> _records)
    {
        records = _records;
    }
    
    // 닉네임 변경
    public void updateNickname(String _nickname)
    {
        nickname = _nickname;
    }
}

File: mrc-server/backend/mrc/src/main/java/com/alpaca/mrc/domain/member/entity/Member.java
package com.alpaca.mrc.domain.member.entity;

import com.alpaca.mrc.domain.record.entity.Record;
import com.alpaca.mrc.domain.shop.entity.MyAvatar;
import com.alpaca.mrc.domain.shop.entity.MyCart;
import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDateTime;
import java.util.List;

@Entity
@Getter
@Setter
@NoArgsConstructor(access = AccessLevel.PROTECTED)
@AllArgsConstructor
@Builder(toBuilder = true)
@Table(name = "member")
public class Member {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @OneToMany(mappedBy = "member", fetch = FetchType.LAZY, cascade = CascadeType.REMOVE)
    private List<Record> records;

    @OneToMany(mappedBy = "member", fetch = FetchType.LAZY, cascade = CascadeType.REMOVE)
    private List<MyCart> myCarts;

    @OneToMany(mappedBy = "member", fetch = FetchType.LAZY, cascade = CascadeType.REMOVE)
    private List<MyAvatar> myAvatars;

    @Column(name = "username", unique = true, nullable = false)
    private String username;

    @Column(name = "nickname", nullable = false)
    private String nickname;

    @Enumerated(EnumType.STRING)
    @Column(name = "account_type", nullable = false)
    private AccountType accountType;

    @Column(name = "icon_url")
    private String iconUrl;

    @Column(name = "coin", columnDefinition = "int default 0")
    private int coin;

    @Column(name = "selected_cart_name")
    private String selectedCartName;

    @Column(name = "selected_avatar_name")
    private String selectedAvatarName;

    @Column(name = "created_at")
    private LocalDateTime createdAt;

    @Column(name = "updated_at")
    private LocalDateTime updatedAt;

    @PrePersist
    public void prePersist() {
        LocalDateTime now = LocalDateTime.now();
        this.createdAt = now;
        this.updatedAt = now;
    }

    @PreUpdate
    public void preUpdate() {
        this.updatedAt = LocalDateTime.now();
    }

    public void updateSelectedCartName(String name) {
        this.selectedCartName = name;
    }

    public void updateSelectedAvatarName(String name) {
        this.selectedAvatarName = name;
    }
}
